### PR TITLE
support compilation on Arch Linux

### PR DESCRIPTION
**Pull request by ccat3z**
### What changes were proposed in this pull request?

Support compilation on Arch Linux. This pr requires #565.

### Why are the changes needed?

Support compilation on Arch Linux.

### Will break the compatibility? How if so?

No.

### Does this PR introduce any user-facing change?

No.

### How was this patch tested?

``` sh
pacman -S git wget rpmextract make binutils
./build.sh release --init --make
```

### Checklist

- [X] I've run the tests to see all new and existing tests pass.
- [X] If this Pull Request resolves an issue, I linked to the issue in the text above.
- [X] I've informed the technical writer about the documentation change if necessary.

### CODE REVIEW

File: deps/3rd/dep_create.sh [hunk 1]
```diff
@@ -65,6 +65,9 @@ function get_os_release() {
       opensuse-leap)
         version_ge "15" && compat_centos7 && return
         ;;
+      arch)
+        compat_centos7 && return
```
Comment (MizuhaHimuraki) [inline deps/3rd/dep_create.sh:69]: version check is needed

**Comment by MizuhaHimuraki** [comment]
https://github.com/oceanbase/oceanbase/wiki/how_to_build
https://github.com/oceanbase/oceanbase#os-compatibility-list

I think you can update the description in these documentations to let developers know that oceanbase is already able to run on Arch Linux.

**Comment by MizuhaHimuraki** [comment]
https://github.com/oceanbase/oceanbase/wiki/how_to_build
https://github.com/oceanbase/oceanbase#os-compatibility-list

I think you can update the description in these documentations to let developers know that oceanbase is already able to run on Arch Linux.

File: deps/3rd/dep_create.sh (hunk 1 above)
Comment (ccat3z) [reply deps/3rd/dep_create.sh:69]: Version check for arch is unnecessary and impossible since arch is a rolling-release system.

Comment (LINxiansheng) [reply deps/3rd/dep_create.sh:69]: Arch Linux use a release date as the os version.

Comment (ccat3z) [reply deps/3rd/dep_create.sh:69]: The release date of Arch Linux is the release date of the installation medium, not the os version.

**Comment by LINxiansheng** [comment]
I fix it in an internal code update and add you to the co-author.
Thanks for your contribution.

**Review by LINxiansheng — APPROVED** [review]

**Comment by LINxiansheng** [comment]
021cb19e6c14c5a9e8ef2aee40e638868d2515aa

**Pull request closed without merging**